src/factotum_server/responder/mod.rs

```diff
@@ -0,0 +1,361 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the Apache License Version 2.0 is distributed on an "AS
+// IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied.  See the Apache License Version 2.0 for the specific language
+// governing permissions and limitations there under.
+//
+
+use std::collections::HashMap;
+use std::error::Error;
+use std::ops::{Deref, DerefMut};
+use std::sync::mpsc;
+use std::sync::mpsc::Sender;
+use iron::mime::*;
+use iron::prelude::*;
+use iron::status;
+use iron::status::Status;
+use url::Url;
+use bodyparser;
+use persistent::{Read, State};
+use serde::Serialize;
+use serde_json;
+
+use factotum_server::{Paths, Server, Storage, Updates};
+use factotum_server::command::Execution;
+use factotum_server::dispatcher::{Dispatch, Query};
+use factotum_server::persistence;
+use factotum_server::persistence::{Persistence, JobState};
+use factotum_server::server::{ServerManager, SettingsRequest, JobRequest};
+
+#[cfg(test)]
+mod tests;
+
+const JSON_CONTENT_TYPE: &'static str = "application/json; charset=UTF-8";
+
+// JSON Response Structs
+
+#[derive(Debug, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct ResponseMessage {
+    message: String
+}
+
+#[derive(Debug, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct FactotumServerStatus {
+    version: VersionStatus,
+    server: ServerStatus,
+    dispatcher: DispatcherStatus,
+}
+
+#[derive(Debug, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct VersionStatus {
+    executor: String
+}
+
+#[derive(Debug, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct ServerStatus {
+    start_time: String,
+    up_time: String,
+    state: String
+}
+
+#[derive(Debug, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct DispatcherStatus {
+    pub workers: WorkerStatus,
+    pub jobs: JobStatus,
+}
+
+#[derive(Debug,PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct WorkerStatus {
+    pub total: usize,
+    pub idle: usize,
+    pub active: usize,
+}
+
+#[derive(Debug, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct JobStatus {
+    pub max_queue_size: usize,
+    pub in_queue: usize,
+    pub fail_count: usize,
+    pub success_count: usize,
+}
+
+// Response handlers
+
+pub fn api(request: &mut Request) -> IronResult<Response> {
+    let url: Url = request.url.clone().into();
+    let response = get_help_message();
+    return_json(status::Ok, encode(&url, response))
+}
+
+pub fn status(request: &mut Request) -> IronResult<Response> {
+    let url: Url = request.url.clone().into();
+    let rwlock = request.get::<State<Server>>().unwrap();
+    let reader = rwlock.read().unwrap();
+    let server_manager = reader.deref();
+    let mutex = request.get::<Read<Updates>>().unwrap();
+    let jobs_channel = mutex.try_lock().unwrap();
```

Comment:
Done - all unwraps replaced with better handling. The ones using expect() could probably be improved.